app/src/main/java/com/example/myapplication/FirstFragment.java

```diff
@@ -27,18 +30,22 @@ public View onCreateView(
             @NonNull LayoutInflater inflater, ViewGroup container,
             Bundle savedInstanceState
     ) {
-        ViewModelProvider viewModelProvider = new ViewModelProvider(requireActivity());
-        UserModel userModel = viewModelProvider.get(UserModel.class);
-        MatchData matchData = new MatchData();
-        userModel.setMatchData(matchData);
         binding = FragmentFirstBinding.inflate(inflater, container, false);
         v = container;
 
-        String currentTeamNumber = UserModel.getMatchData().getTeamNumber();
-        binding.input.setText(currentTeamNumber);
-
         String currentMatchNumber = UserModel.getMatchData().getMatchNumber();
         binding.matchInput.setText(currentMatchNumber);
+        try {
+            String currentTeamNumber = MainActivity.teams.get(Integer.parseInt(currentMatchNumber) - 1);
+            binding.input.setText(currentTeamNumber);
+        } catch (Exception e){
+
```

Comment:
It's not a good idea to swallow/ignore exceptions. Is there something reasonable you can do here if there's an error? Maybe pop open a dialog box for the user, or at least output a debug message somewhere?